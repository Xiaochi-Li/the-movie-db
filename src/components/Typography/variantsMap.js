import theme from "../../assets/theme";

const { color, font } = theme;
const { family, size, lineHeight, weight } = font;

export default new Map([
  /**
   * Header 1 - 6
   * These are rendered as their normal tags
   * They use header font
   */
  [
    "h1",
    {
      style: {
        fontSize: size.xlg,
        fontWeight: weight.bold,
        lineHeight: lineHeight.xlg,
        fontFamily: family.montserrat,
        color: color.text.light
      },
      as: "h1"
    }
  ],
  [
    "h2",
    {
      style: {
        fontSize: size.lg,
        fontWeight: weight.bold,
        lineHeight: lineHeight.lg,
        fontFamily: family.montserrat,
        color: color.text.light
      },
      as: "h2"
    }
  ],
  [
    "h3",
    {
      style: {
        fontSize: size.sm,
        fontWeight: weight.normal,
        lineHeight: lineHeight.md,
        fontFamily: family.roboto,
        color: color.text.mdLight
      },
      as: "h3"
    }
  ],
  [
    "body",
    {
      style: {
        fontSize: size.md,
        fontWeight: weight.normal,
        lineHeight: lineHeight.lg,
        fontFamily: family.roboto,
        color: color.text.mdGrey
      },
      as: "p"
    }
  ],
  [
    "caption1",
    {
      style: {
        fontSize: size.xsm,
        fontWeight: weight.normal,
        lineHeight: lineHeight.sm,
        fontFamily: family.roboto,
        color: color.text.lightGrey
      },
      as: "span"
    }
  ],
  [
    "caption2",
    {
      style: {
        fontSize: size.xsm,
        fontWeight: weight.normal,
        lineHeight: lineHeight.sm,
        fontFamily: family.roboto,
        color: color.text.grey
      },
      as: "span"
    }
  ],
  [
    "caption3",
    {
      style: {
        fontSize: size.xsm,
        fontWeight: weight.normal,
        lineHeight: lineHeight.sm,
        fontFamily: family.roboto,
        textAlign: "center",
        color: color.text.white
      },
      as: "span"
    }
  ]
]);
