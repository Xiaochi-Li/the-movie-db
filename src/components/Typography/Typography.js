import React from "react";
import { withTheme } from "styled-components";
import PropTypes from "prop-types";

import ThemePropTypes from "../../assets/ThemePropTypes"
import variantsMap from "./variantsMap";
import { observer } from "mobx-react";

const Typography = ({ variant, color, theme, ...otherProps }) => {
  // Get correct variant or default to body - PropTypes will throw a warning if wrong
  const selectedVariant = variantsMap.get(variant) || variantsMap.get("body");
  // Decide what html tag to use. Use the provided, or the variants default
  const HTMLTag = selectedVariant.as;
  return <HTMLTag {...otherProps} style={selectedVariant.style} />;
};

Typography.defaultProps = {
  variant: "body"
};

Typography.propTypes = {
  variant: PropTypes.oneOf([...variantsMap.keys()]),
  theme: ThemePropTypes.isRequired
};

export default withTheme(observer(Typography));
