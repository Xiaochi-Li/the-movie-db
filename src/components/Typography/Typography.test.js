import React from "react";
import variantsMap from "./variantsMap";
import renderer from "react-test-renderer";
import Typography from "./Typography";

test("Can create Typography", () => {
  expect(Typography).toBeDefined();
  variantsMap.forEach((variantValue, variantKey) => {
    const component = renderer.create(
      <Typography varian={variantKey}>Test</Typography>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

test("Can override default custom css", () => {
  const customStyle = {
    color: "red",
    fontSize: "20px"
  };
  const component = renderer.create(
    <Typography variant="h1" customizeStyle={customStyle}>
      Styling Test
    </Typography>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
