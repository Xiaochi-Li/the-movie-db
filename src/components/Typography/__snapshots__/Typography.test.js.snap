// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Can create Typography 1`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="h1"
>
  Test
</p>
`;

exports[`Can create Typography 2`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="h2"
>
  Test
</p>
`;

exports[`Can create Typography 3`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="h3"
>
  Test
</p>
`;

exports[`Can create Typography 4`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="body"
>
  Test
</p>
`;

exports[`Can create Typography 5`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="caption1"
>
  Test
</p>
`;

exports[`Can create Typography 6`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="caption2"
>
  Test
</p>
`;

exports[`Can create Typography 7`] = `
<p
  style={
    Object {
      "color": "#9FBBC7",
      "fontFamily": "Roboto",
      "fontSize": "16px",
      "fontWeight": "normal",
      "lineHeight": "24px",
    }
  }
  varian="caption3"
>
  Test
</p>
`;

exports[`Can override default custom css 1`] = `
<h1
  style={
    Object {
      "color": "#E3F4FC",
      "fontFamily": " 'Montserrat', sans-serif",
      "fontSize": "28px",
      "fontWeight": "bold",
      "lineHeight": "30px",
      "margin": "12px 0",
    }
  }
>
  Styling Test
</h1>
`;
